lib/path.js

```diff
@@ -1073,16 +1073,13 @@ const posix = {
   resolve: function resolve() {
     var resolvedPath = '';
     var resolvedAbsolute = false;
-    var cwd;
 
     for (var i = arguments.length - 1; i >= -1 && !resolvedAbsolute; i--) {
       var path;
       if (i >= 0)
         path = arguments[i];
       else {
-        if (cwd === undefined)
-          cwd = process.cwd();
```

Comment:
I think this is a performance optimization to avoid calling `process.cwd()` more than once.

Reply:
I don't think `process.cwd` would be called more then once since it only executes when the variable `I` is equal to -1 (less than 0).